src/solvers/refinement/string_refinement_util.h

```diff
@@ -297,6 +297,45 @@ class dependency_grapht
   /// Nodes describing dependencies of a string: values of the map correspond
   /// to indexes in the vector `string_nodes`.
   std::map<array_string_exprt, std::size_t> node_index_pool;
+
+  /// Common index for all nodes (both strings and primitives) so that we can
+  /// reuse generic algorithms of util/graph.h
+  /// Even indexes correspond to primitive nodes, odd indexes to string nodes.
+  typedef std::size_t node_indext;
+
+  node_indext size() const
+  {
+    return primitive_nodes.size() + string_nodes.size();
+  }
+
+  node_indext node_index(const primitive_nodet &p) const
+  {
+    return 2 * p.index;
+  }
+
+  node_indext node_index(const array_string_exprt &s) const
```

Comment:
What is `node_index`? and why is it a function of the size of `primitive_nodes`?